core/src/main/scala/org/json4s/Extraction.scala

```diff
@@ -572,6 +572,16 @@ object Extraction {
     private[this] def buildCtorArg(json: JValue, descr: ConstructorParamDescriptor) = {
       val default = descr.defaultValue
       def defv(v: Any) = if (default.isDefined) default.get() else v
+
+      // Test that uses custom serialize for null to give None for optional values
+      def nullCheck(j: JValue): Boolean = {
+        try {
+          extract(j, ScalaType[Null](implicitly)) == null
+        } catch {
+          case _: Throwable => false
```

Comment:
Can we go with `case scala.util.control.NonFatal(_) => false` here?

Reply:
As you pointed out, it is better to limit to NonFatal. I will fix.